Affinity (#1993)
spinnaker/spinnaker.github.io: reference/halyard/custom.md

| `imagePullSecrets` | Define secrets to use to pull a custom artifact when using an artifactId to swap out docker image. More details can be found in the [Image Pull Secrets section](#imagepullsecrets). |
| `nodePort` | When using a serviceType of NodePort, this provides the NodePort value. |
| `nodeSelectors` | Provide a list of `nodeSelectors` key-value pairs to add to the pod specification. See [Node Selectors](#nodeselectors) below. |

> | `affinity` | Provide a list of `affinity` per Spinnaker component to add to the pod specification. See [Affinity](#affinity) below. |

Reviewer: "...list of affinity _rules_"?

Reviewer: or "affinity annotations"?

Reviewer 2: `affinity or anti-affinity rules`

Reviewer 2: ```Provide a list of `affinity` rules applied to a Spinnaker components pod specification.```

Author: Committed: https://github.com/spinnaker/spinnaker.github.io/pull/1993/commits/98742d96427f62abe52c575352ae62f7f7e58c3f

| `imagePullSecrets` | Define secrets to use to pull a custom artifact when using an artifactId to swap out docker image. More details can be found in the [Image Pull Secrets section](#imagepullsecrets). |
| `nodePort` | When using a serviceType of NodePort, this provides the NodePort value. |
| `nodeSelectors` | Provide a list of `nodeSelectors` key-value pairs to add to the pod specification. See [Node Selectors](#nodeselectors) below. |

> | `affinity` | Provide a list of `affinity` rules applied to a Spinnaker components pod specification. See [Affinity](#affinity) below. |

Reviewer: ```suggestion
| `affinity` | Provide a list of `affinity` rules applied to a Spinnaker component's pod specification. See [Affinity](#affinity) below. |
```

```
kubernetes:
  deploymentEnvironment:
    nodeSelectors:
      exampleNodeKey: exampleNodeValue
```

Additional information regarding Node Selectors can be found in [the Kubernetes NodeSelector documentation](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector).

### affinity

> Affinity annotations will put out `affinity` values in the Pod specification and is more expressive than `nodeSelector`. Precedence can be set rather than a hard requirement. The two types are `node affinity` and `inter-pod affinity/anti-affinity`.

Reviewer: ```suggestion
Affinity annotations generate `affinity` values in the Pod specification and are more expressive than `nodeSelector`. You can set precedence rather than specify an affinity hard requirement. The two types are `node affinity` and `inter-pod affinity/anti-affinity`.
```

Reviewer 2: ```Affinity annotations may apply one or many `affinity` or `anti-affinity` rules to a Pod specification, allowing additional flexibility over `nodeSelector`.```

Provide configuration per spinnaker component like the below example.

```
kubernetes:
  deploymentEnvironment:
    affinity:
      spin-clouddriver:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: spinnaker
                operator: In
                values:
                - 'true'
```

> Additional information regarding `affinity` can be found in [the Kubernetes affinity documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity).

Reviewer: ```suggestion
See [the Kubernetes affinity documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity) for more information.
```

Reviewer 2: like the change
